src/codeu/chat/server/LocalFile.java

```diff
@@ -0,0 +1,166 @@
+package codeu.chat.server;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Serializable;
+import java.io.ObjectOutputStream;
+import java.io.RandomAccessFile;
+import java.io.FileOutputStream;
+import java.io.ObjectInputStream;
+import java.io.FileInputStream;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import codeu.chat.common.ConversationHeader;
+import codeu.chat.common.Message;
+import codeu.chat.common.User;
+import codeu.chat.util.Logger;
+import codeu.chat.util.Uuid;
+
+public class LocalFile implements Serializable
+{
+    private static final long serialVersionUID = 1L;
+    //Instance varibles for saving the current data of server.
+    private final HashMap<Uuid,User> users;
+    private final HashMap<Uuid,ConversationHeader> conversationHeaders;
+    private final HashMap<Uuid,Message> messages;
+
+    private transient final File file;
+
+    private static final Logger.Log LOG = Logger.newLog(LocalFile.class);
+
+    private transient boolean hasModified = false;//It indicates if there is a new data should be handled.
+    public LocalFile(File file)
+    {
+        this.file = file;
+        LocalFile newClass;
+        HashMap<Uuid,User>tempUsers = new HashMap<>();
+        HashMap<Uuid,ConversationHeader>tempConversationHeaders= new HashMap<>();
+        HashMap<Uuid,Message>tempMessages= new HashMap<>();
+        try
+        {
+            //Deserialization
+            RandomAccessFile randomAccessFile = new RandomAccessFile(file, "r");
+            FileInputStream fileInputStream = new FileInputStream(randomAccessFile.getFD());
+		    ObjectInputStream objectInputStream = new ObjectInputStream(fileInputStream);
+            //Read the information to current instance
+		    newClass = (LocalFile)objectInputStream.readObject();
+            tempUsers = newClass.users;
+            tempConversationHeaders = newClass.conversationHeaders;
+            tempMessages = newClass.messages;
+
+            fileInputStream.close();
+            objectInputStream.close();
+        }
+        catch(Exception exception)
+        {
+            System.out.println("ERROR:Unable to read data.");
+            System.out.println("WARNING:If it is the case that the file doesn't exist, a new local file will be created.");
+            exception.printStackTrace();
+        }
+        finally
+        {
+            //Write temporary information to the instance varibles
+            users = tempUsers;
```

Comment:
I don't think this `finally` block is a good idea. If the code in the `try` block fails, tempUsers, tempConversationHeaders, and tempMessages will all be null.  If that happens, the program will enter a weird state, where it could potentially overwrite valid users/conversations/messages in the file the next time that it saves. 

In my opinion, you should remove this `finally` block.  If there are exceptions above, you should allow the program to crash rather than assigning these objects to null.